logstash-core/spec/logstash/agent/converge_spec.rb

```diff
@@ -191,7 +191,7 @@
           end
 
           it "it will keep trying to converge" do
-            sleep(agent_settings.get("config.reload.interval").to_seconds * 20) # let the interval reload a few times
+            sleep(agent_settings.get("config.reload.interval").to_nanos * 20 / 1_000_000_000.0) # let the interval reload a few times
```

Comment:
```suggestion
            # we can't do .to_seconds here as values under 1 seconds are rounded to 0
            # causing a race condition in the test. So we get the nanos and convert to seconds
            sleep(agent_settings.get("config.reload.interval").to_nanos * 1e-9 * 20) # let the interval reload a few times
```

Reply:
I've made this change, and added a similar fix to `agent.rb`